packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -475,13 +479,30 @@ processors:
   - urldecode:
       field: _tmp.cs_headers
       ignore_missing: true
-  - kv:
-      field: _tmp.cs_headers
-      description: Parse request headers string into individual header fields.
-      target_field: http.request.headers
-      field_split: '\n'
-      value_split: ':'
-      ignore_missing: true
+  - script:
+      lang: painless
+      description: This script parses and standardizes the http headers
+      source: |
+        String header_string = ctx._tmp.cs_headers;
+        if (header_string != null) {
+          header_string = header_string.trim();
+          Map header_map = new HashMap();
+
+          // https://gist.github.com/vjt/06b28fbd988788c2a7a71c63dd9163be
+          def header_list = header_string.splitOnToken((String)(char)0x0a);
+
+          for (item in header_list) {
+            def kv = item.splitOnToken(':', 2);
+            if (kv.length == 2) {
+              String key = kv[0].toLowerCase().replace('-','_');
```

Comment:
@boecklic why changing he header from hyphen to underscore ? Might be a good idea to comment the reason here. Perso I would have kept hyphen for headers but you surely have a good reason for this ?

Reply:
The idea was to use underlines, like recommended on other field names. But technically it is not necessary. I can easily revert that.